PersistenceMigrationsSample/app/src/room2/java/com/example/android/persistence/migrations/UsersDatabase.java

```diff
@@ -67,8 +67,15 @@ public void migrate(SupportSQLiteDatabase database) {
     static final Migration MIGRATION_2_3 = new Migration(2, 3) {
         @Override
         public void migrate(SupportSQLiteDatabase database) {
-            database.execSQL("ALTER TABLE Users "
-                    + " ADD COLUMN last_update INTEGER");
+            database.execSQL("CREATE TABLE IF NOT EXISTS Users_new ("
```

Comment:
Leave this as it was since the version 2 already had NON NULL primary key and the only update from version 2 to 3 is the addition of a new column.

Reply:
hmm i should double check why this was necessary to pass, maybe i did it in reverse order.